File: 刷题/cpp/LCP 50. 宝石补给.cpp
class Solution {
public:
    int giveGem(vector<int>& gem, vector<vector<int>>& operations) {
        for(auto operation: operations){
            int x=operation[0],y=operation[1];
            int z=gem[x]/2;
            gem[y]+=z;
            gem[x]-=z;
        }
        sort(gem.begin(),gem.end());
        return gem[gem.size()-1]-gem[0];
    }
};

File: 刷题/golang/LCP 50. 宝石补给.go
func giveGem(gem []int, operations [][]int) int {
    for _, operation := range operations {
        x, y := operation[0], operation[1]
        number:=gem[x]/2
        gem[y]+=number
        gem[x]-=number
    }
    mn, mx := gem[0], gem[0]
    for _, number := range gem {
        mn = min(number, mn)
        mx = max(number, mx)
    }
    return mx - mn
}

func max(x int, y int) int {
    if x < y {
        return y
    }
    return x
}

func min(x int, y int) int {
    if x > y {
        return y
    }
    return x
}
